File: patterns/chat-basics/gemini/javascript/basic-chat.ts
import 'dotenv/config';
import { GoogleGenerativeAI } from '@google/generative-ai';

/**
 * Demonstrates a basic chat interaction using Google's Gemini API.
 */
async function main() {
    const apiKey = process.env.GEMINI_API_KEY;
    if (!apiKey) {
        console.error('Error: GEMINI_API_KEY not found in environment variables.');
        return;
    }

    // Initialize the client
    const genAI = new GoogleGenerativeAI(apiKey);

    console.log('Sending request to Gemini...');

    try {
        // Get the model
        const model = genAI.getGenerativeModel({ model: 'gemini-1.5-flash' });

        // Generate content
        const result = await model.generateContent('Explain quantum computing in one sentence.');
        const response = await result.response;
        const text = response.text();

        console.log('\nChat response:');
        console.log(text);
    } catch (error) {
        console.error('Error calling Gemini API:', error);
    }
}

main();

File: patterns/chat-basics/gemini/python/basic_chat.py
import os
import google.generativeai as genai
from dotenv import load_dotenv

# Load environment variables
load_dotenv()

def main():
    """
    Demonstrates a basic chat interaction using Google's Gemini API.
    """
    api_key = os.environ.get("GEMINI_API_KEY")
    if not api_key:
        print("Error: GEMINI_API_KEY not found in environment variables.")
        return

    # Configure the library
    genai.configure(api_key=api_key)

    print("Sending request to Gemini...")

    # Initialize the model
    model = genai.GenerativeModel('gemini-1.5-flash')

    # Generate content (single turn)
    response = model.generate_content("Explain quantum computing in one sentence.")

    print("\nChat response:")
    print(response.text)

if __name__ == "__main__":
    main()
